{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we start looking at the coverage, we want to first import the necessary modules. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "\n",
    "from itertools import islice \n",
    "from sklearn.cluster import KMeans"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, we want to define a function that joins the directories of the desired path in hopes of simplifying the process of reading the .tsv files. In this function, we want to make sure the inputted file is a string. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def join_file(__file__): #\n",
    "    dir = r'~/Coverage/data/12_SUMMARY/bin_by_bin/ALL_SPLITS'\n",
    "    return os.path.join(dir, __file__)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we load up the .tsv files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The coverage values for each metagenome with specific genes\n",
    "df = pd.read_csv(join_file('ALL_SPLITS-gene_non_outlier_coverages.txt'), \n",
    "                 sep = '\\t').set_index('gene_callers_id')\n",
    "\n",
    "# The standard deviation coverage values \n",
    "df_sigma = pd.read_csv(join_file('ALL_SPLITS-gene_non_outlier_coverage_stds.txt'), \n",
    "                        sep = '\\t').set_index('gene_callers_id')\n",
    "\n",
    "\n",
    "data_dir = r'../data/'\n",
    "df.to_csv(data_dir + 'Coverage.csv')\n",
    "df_sigma.to_csv(data_dir +'Coverage_sigma.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to see what columns we are working with, so we print the first 5 rows of our dataframe. Moreover, we also want a basic summary of the rudimentary statistical values."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ANE_004_05M</th>\n",
       "      <th>ANE_004_40M</th>\n",
       "      <th>ANE_150_05M</th>\n",
       "      <th>ANE_150_40M</th>\n",
       "      <th>ANE_151_05M</th>\n",
       "      <th>ANE_151_80M</th>\n",
       "      <th>ANE_152_05M</th>\n",
       "      <th>ANW_141_05M</th>\n",
       "      <th>ANW_142_05M</th>\n",
       "      <th>ANW_145_05M</th>\n",
       "      <th>...</th>\n",
       "      <th>PSW_128_40M</th>\n",
       "      <th>RED_31_05M</th>\n",
       "      <th>RED_32_05M</th>\n",
       "      <th>RED_32_80M</th>\n",
       "      <th>RED_33_05M</th>\n",
       "      <th>RED_34_05M</th>\n",
       "      <th>RED_34_60M</th>\n",
       "      <th>SOC_84_05M</th>\n",
       "      <th>SOC_85_05M</th>\n",
       "      <th>SOC_85_90M</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_callers_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1230</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.701389</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.955056</td>\n",
       "      <td>1.082962</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.044402</td>\n",
       "      <td>1.196158</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.959543</td>\n",
       "      <td>0.920139</td>\n",
       "      <td>3.242754</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.801282</td>\n",
       "      <td>7.103571</td>\n",
       "      <td>0.885068</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1231</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.779553</td>\n",
       "      <td>1.075638</td>\n",
       "      <td>1.009675</td>\n",
       "      <td>0.640091</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.894249</td>\n",
       "      <td>1.137011</td>\n",
       "      <td>0.745358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.985728</td>\n",
       "      <td>0.502580</td>\n",
       "      <td>0.631198</td>\n",
       "      <td>0.895377</td>\n",
       "      <td>3.016057</td>\n",
       "      <td>1.743184</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1232</td>\n",
       "      <td>2.308204</td>\n",
       "      <td>8.848606</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>2.940000</td>\n",
       "      <td>4.534161</td>\n",
       "      <td>4.615087</td>\n",
       "      <td>3.880342</td>\n",
       "      <td>4.780449</td>\n",
       "      <td>0.566667</td>\n",
       "      <td>1.901376</td>\n",
       "      <td>...</td>\n",
       "      <td>3.362563</td>\n",
       "      <td>2.746753</td>\n",
       "      <td>5.094574</td>\n",
       "      <td>0.656842</td>\n",
       "      <td>6.289425</td>\n",
       "      <td>6.549606</td>\n",
       "      <td>0.717694</td>\n",
       "      <td>1.008104</td>\n",
       "      <td>1.869355</td>\n",
       "      <td>1.077206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1233</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.673582</td>\n",
       "      <td>1.717138</td>\n",
       "      <td>5.044211</td>\n",
       "      <td>1.370543</td>\n",
       "      <td>1.016949</td>\n",
       "      <td>1.592806</td>\n",
       "      <td>1.721739</td>\n",
       "      <td>0.838174</td>\n",
       "      <td>...</td>\n",
       "      <td>1.092669</td>\n",
       "      <td>0.920128</td>\n",
       "      <td>2.012346</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.829588</td>\n",
       "      <td>4.966741</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1234</td>\n",
       "      <td>0.370583</td>\n",
       "      <td>1.080229</td>\n",
       "      <td>1.181025</td>\n",
       "      <td>1.667647</td>\n",
       "      <td>2.080544</td>\n",
       "      <td>2.748344</td>\n",
       "      <td>0.713467</td>\n",
       "      <td>2.011351</td>\n",
       "      <td>3.738072</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.802871</td>\n",
       "      <td>1.115991</td>\n",
       "      <td>1.375427</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>9.217054</td>\n",
       "      <td>7.203260</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.086915</td>\n",
       "      <td>0.203438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 ANE_004_05M  ANE_004_40M  ANE_150_05M  ANE_150_40M  \\\n",
       "gene_callers_id                                                       \n",
       "1230                0.000000     0.000000     0.701389     0.000000   \n",
       "1231                0.000000     0.779553     1.075638     1.009675   \n",
       "1232                2.308204     8.848606     0.800000     2.940000   \n",
       "1233                0.000000     0.000000     1.673582     1.717138   \n",
       "1234                0.370583     1.080229     1.181025     1.667647   \n",
       "\n",
       "                 ANE_151_05M  ANE_151_80M  ANE_152_05M  ANW_141_05M  \\\n",
       "gene_callers_id                                                       \n",
       "1230                2.955056     1.082962     0.000000     2.044402   \n",
       "1231                0.640091     0.000000     0.000000     0.894249   \n",
       "1232                4.534161     4.615087     3.880342     4.780449   \n",
       "1233                5.044211     1.370543     1.016949     1.592806   \n",
       "1234                2.080544     2.748344     0.713467     2.011351   \n",
       "\n",
       "                 ANW_142_05M  ANW_145_05M  ...  PSW_128_40M  RED_31_05M  \\\n",
       "gene_callers_id                            ...                            \n",
       "1230                1.196158     0.000000  ...     0.959543    0.920139   \n",
       "1231                1.137011     0.745358  ...     0.000000    0.985728   \n",
       "1232                0.566667     1.901376  ...     3.362563    2.746753   \n",
       "1233                1.721739     0.838174  ...     1.092669    0.920128   \n",
       "1234                3.738072     0.000000  ...     1.802871    1.115991   \n",
       "\n",
       "                 RED_32_05M  RED_32_80M  RED_33_05M  RED_34_05M  RED_34_60M  \\\n",
       "gene_callers_id                                                               \n",
       "1230               3.242754    0.000000    1.801282    7.103571    0.885068   \n",
       "1231               0.502580    0.631198    0.895377    3.016057    1.743184   \n",
       "1232               5.094574    0.656842    6.289425    6.549606    0.717694   \n",
       "1233               2.012346    0.000000    3.829588    4.966741    0.000000   \n",
       "1234               1.375427    0.000000    9.217054    7.203260    0.000000   \n",
       "\n",
       "                 SOC_84_05M  SOC_85_05M  SOC_85_90M  \n",
       "gene_callers_id                                      \n",
       "1230               0.000000    0.000000    0.000000  \n",
       "1231               0.000000    0.000000    0.000000  \n",
       "1232               1.008104    1.869355    1.077206  \n",
       "1233               0.000000    0.000000    0.000000  \n",
       "1234               0.000000    0.086915    0.203438  \n",
       "\n",
       "[5 rows x 93 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ANE_004_05M</th>\n",
       "      <th>ANE_004_40M</th>\n",
       "      <th>ANE_150_05M</th>\n",
       "      <th>ANE_150_40M</th>\n",
       "      <th>ANE_151_05M</th>\n",
       "      <th>ANE_151_80M</th>\n",
       "      <th>ANE_152_05M</th>\n",
       "      <th>ANW_141_05M</th>\n",
       "      <th>ANW_142_05M</th>\n",
       "      <th>ANW_145_05M</th>\n",
       "      <th>...</th>\n",
       "      <th>PSW_128_40M</th>\n",
       "      <th>RED_31_05M</th>\n",
       "      <th>RED_32_05M</th>\n",
       "      <th>RED_32_80M</th>\n",
       "      <th>RED_33_05M</th>\n",
       "      <th>RED_34_05M</th>\n",
       "      <th>RED_34_60M</th>\n",
       "      <th>SOC_84_05M</th>\n",
       "      <th>SOC_85_05M</th>\n",
       "      <th>SOC_85_90M</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>count</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mean</td>\n",
       "      <td>172.817982</td>\n",
       "      <td>299.608349</td>\n",
       "      <td>321.301558</td>\n",
       "      <td>275.781062</td>\n",
       "      <td>308.886857</td>\n",
       "      <td>216.603589</td>\n",
       "      <td>66.355116</td>\n",
       "      <td>572.462331</td>\n",
       "      <td>290.120408</td>\n",
       "      <td>228.006186</td>\n",
       "      <td>...</td>\n",
       "      <td>219.122153</td>\n",
       "      <td>466.189768</td>\n",
       "      <td>505.800545</td>\n",
       "      <td>248.588116</td>\n",
       "      <td>723.774598</td>\n",
       "      <td>638.401231</td>\n",
       "      <td>99.863169</td>\n",
       "      <td>1.103168</td>\n",
       "      <td>1.295401</td>\n",
       "      <td>1.196534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>std</td>\n",
       "      <td>154.800781</td>\n",
       "      <td>235.618106</td>\n",
       "      <td>244.113452</td>\n",
       "      <td>212.290976</td>\n",
       "      <td>239.216560</td>\n",
       "      <td>168.868690</td>\n",
       "      <td>72.272410</td>\n",
       "      <td>375.789115</td>\n",
       "      <td>233.248835</td>\n",
       "      <td>160.110231</td>\n",
       "      <td>...</td>\n",
       "      <td>244.194078</td>\n",
       "      <td>361.228508</td>\n",
       "      <td>377.690965</td>\n",
       "      <td>193.867680</td>\n",
       "      <td>502.582064</td>\n",
       "      <td>433.403013</td>\n",
       "      <td>75.818548</td>\n",
       "      <td>8.583339</td>\n",
       "      <td>8.857377</td>\n",
       "      <td>8.053501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>min</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25%</td>\n",
       "      <td>30.114968</td>\n",
       "      <td>54.623854</td>\n",
       "      <td>61.933014</td>\n",
       "      <td>49.221619</td>\n",
       "      <td>58.852271</td>\n",
       "      <td>40.301266</td>\n",
       "      <td>15.705094</td>\n",
       "      <td>167.960504</td>\n",
       "      <td>34.249639</td>\n",
       "      <td>64.697330</td>\n",
       "      <td>...</td>\n",
       "      <td>12.027030</td>\n",
       "      <td>70.202304</td>\n",
       "      <td>86.746732</td>\n",
       "      <td>32.206848</td>\n",
       "      <td>168.894631</td>\n",
       "      <td>158.107415</td>\n",
       "      <td>16.531513</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50%</td>\n",
       "      <td>172.952461</td>\n",
       "      <td>305.426285</td>\n",
       "      <td>333.545054</td>\n",
       "      <td>280.840187</td>\n",
       "      <td>311.853661</td>\n",
       "      <td>219.560387</td>\n",
       "      <td>53.983524</td>\n",
       "      <td>681.784998</td>\n",
       "      <td>291.791218</td>\n",
       "      <td>249.307854</td>\n",
       "      <td>...</td>\n",
       "      <td>117.841391</td>\n",
       "      <td>490.641618</td>\n",
       "      <td>564.819490</td>\n",
       "      <td>263.568268</td>\n",
       "      <td>843.046406</td>\n",
       "      <td>755.055252</td>\n",
       "      <td>104.968085</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75%</td>\n",
       "      <td>260.667045</td>\n",
       "      <td>469.651054</td>\n",
       "      <td>507.411206</td>\n",
       "      <td>436.256089</td>\n",
       "      <td>483.108702</td>\n",
       "      <td>338.482581</td>\n",
       "      <td>91.291407</td>\n",
       "      <td>880.165387</td>\n",
       "      <td>484.011622</td>\n",
       "      <td>353.933527</td>\n",
       "      <td>...</td>\n",
       "      <td>371.736583</td>\n",
       "      <td>765.930510</td>\n",
       "      <td>845.990101</td>\n",
       "      <td>408.179149</td>\n",
       "      <td>1149.984970</td>\n",
       "      <td>1009.565833</td>\n",
       "      <td>162.233696</td>\n",
       "      <td>0.546757</td>\n",
       "      <td>0.619354</td>\n",
       "      <td>0.687012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>max</td>\n",
       "      <td>2300.918919</td>\n",
       "      <td>1355.894118</td>\n",
       "      <td>1337.613333</td>\n",
       "      <td>1178.283333</td>\n",
       "      <td>1592.222222</td>\n",
       "      <td>1078.027778</td>\n",
       "      <td>804.506667</td>\n",
       "      <td>1290.526480</td>\n",
       "      <td>855.790896</td>\n",
       "      <td>857.325301</td>\n",
       "      <td>...</td>\n",
       "      <td>1054.708767</td>\n",
       "      <td>1511.473118</td>\n",
       "      <td>1217.336538</td>\n",
       "      <td>688.960976</td>\n",
       "      <td>1690.004000</td>\n",
       "      <td>1466.757202</td>\n",
       "      <td>281.294118</td>\n",
       "      <td>305.910053</td>\n",
       "      <td>308.513196</td>\n",
       "      <td>279.843305</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ANE_004_05M  ANE_004_40M  ANE_150_05M  ANE_150_40M  ANE_151_05M  \\\n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000   \n",
       "mean    172.817982   299.608349   321.301558   275.781062   308.886857   \n",
       "std     154.800781   235.618106   244.113452   212.290976   239.216560   \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000   \n",
       "25%      30.114968    54.623854    61.933014    49.221619    58.852271   \n",
       "50%     172.952461   305.426285   333.545054   280.840187   311.853661   \n",
       "75%     260.667045   469.651054   507.411206   436.256089   483.108702   \n",
       "max    2300.918919  1355.894118  1337.613333  1178.283333  1592.222222   \n",
       "\n",
       "       ANE_151_80M  ANE_152_05M  ANW_141_05M  ANW_142_05M  ANW_145_05M  ...  \\\n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000  ...   \n",
       "mean    216.603589    66.355116   572.462331   290.120408   228.006186  ...   \n",
       "std     168.868690    72.272410   375.789115   233.248835   160.110231  ...   \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
       "25%      40.301266    15.705094   167.960504    34.249639    64.697330  ...   \n",
       "50%     219.560387    53.983524   681.784998   291.791218   249.307854  ...   \n",
       "75%     338.482581    91.291407   880.165387   484.011622   353.933527  ...   \n",
       "max    1078.027778   804.506667  1290.526480   855.790896   857.325301  ...   \n",
       "\n",
       "       PSW_128_40M   RED_31_05M   RED_32_05M   RED_32_80M   RED_33_05M  \\\n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000   \n",
       "mean    219.122153   466.189768   505.800545   248.588116   723.774598   \n",
       "std     244.194078   361.228508   377.690965   193.867680   502.582064   \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000   \n",
       "25%      12.027030    70.202304    86.746732    32.206848   168.894631   \n",
       "50%     117.841391   490.641618   564.819490   263.568268   843.046406   \n",
       "75%     371.736583   765.930510   845.990101   408.179149  1149.984970   \n",
       "max    1054.708767  1511.473118  1217.336538   688.960976  1690.004000   \n",
       "\n",
       "        RED_34_05M   RED_34_60M   SOC_84_05M   SOC_85_05M   SOC_85_90M  \n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000  \n",
       "mean    638.401231    99.863169     1.103168     1.295401     1.196534  \n",
       "std     433.403013    75.818548     8.583339     8.857377     8.053501  \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000  \n",
       "25%     158.107415    16.531513     0.000000     0.000000     0.000000  \n",
       "50%     755.055252   104.968085     0.000000     0.000000     0.000000  \n",
       "75%    1009.565833   162.233696     0.546757     0.619354     0.687012  \n",
       "max    1466.757202   281.294118   305.910053   308.513196   279.843305  \n",
       "\n",
       "[8 rows x 93 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The same logic is applied to the standard deviation coverage values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ANE_004_05M</th>\n",
       "      <th>ANE_004_40M</th>\n",
       "      <th>ANE_150_05M</th>\n",
       "      <th>ANE_150_40M</th>\n",
       "      <th>ANE_151_05M</th>\n",
       "      <th>ANE_151_80M</th>\n",
       "      <th>ANE_152_05M</th>\n",
       "      <th>ANW_141_05M</th>\n",
       "      <th>ANW_142_05M</th>\n",
       "      <th>ANW_145_05M</th>\n",
       "      <th>...</th>\n",
       "      <th>PSW_128_40M</th>\n",
       "      <th>RED_31_05M</th>\n",
       "      <th>RED_32_05M</th>\n",
       "      <th>RED_32_80M</th>\n",
       "      <th>RED_33_05M</th>\n",
       "      <th>RED_34_05M</th>\n",
       "      <th>RED_34_60M</th>\n",
       "      <th>SOC_84_05M</th>\n",
       "      <th>SOC_85_05M</th>\n",
       "      <th>SOC_85_90M</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_callers_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1230</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.721253</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.155447</td>\n",
       "      <td>1.070772</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.307078</td>\n",
       "      <td>1.116991</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.789185</td>\n",
       "      <td>1.077167</td>\n",
       "      <td>2.026755</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.959566</td>\n",
       "      <td>1.298346</td>\n",
       "      <td>0.971263</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1231</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.033706</td>\n",
       "      <td>1.632857</td>\n",
       "      <td>1.407635</td>\n",
       "      <td>0.909497</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.811527</td>\n",
       "      <td>1.065848</td>\n",
       "      <td>0.759649</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.144542</td>\n",
       "      <td>0.642688</td>\n",
       "      <td>0.84632</td>\n",
       "      <td>1.154175</td>\n",
       "      <td>2.702909</td>\n",
       "      <td>1.713162</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1232</td>\n",
       "      <td>4.528344</td>\n",
       "      <td>7.415190</td>\n",
       "      <td>1.292512</td>\n",
       "      <td>3.206930</td>\n",
       "      <td>4.631825</td>\n",
       "      <td>5.682169</td>\n",
       "      <td>3.922132</td>\n",
       "      <td>3.784440</td>\n",
       "      <td>0.775314</td>\n",
       "      <td>1.938717</td>\n",
       "      <td>...</td>\n",
       "      <td>2.257851</td>\n",
       "      <td>1.992068</td>\n",
       "      <td>4.024967</td>\n",
       "      <td>0.84488</td>\n",
       "      <td>2.544174</td>\n",
       "      <td>2.909985</td>\n",
       "      <td>1.034122</td>\n",
       "      <td>1.023991</td>\n",
       "      <td>1.681214</td>\n",
       "      <td>0.912291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1233</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.376027</td>\n",
       "      <td>1.012331</td>\n",
       "      <td>1.112675</td>\n",
       "      <td>1.068535</td>\n",
       "      <td>0.935034</td>\n",
       "      <td>1.281728</td>\n",
       "      <td>0.915704</td>\n",
       "      <td>0.872245</td>\n",
       "      <td>...</td>\n",
       "      <td>0.941432</td>\n",
       "      <td>0.950875</td>\n",
       "      <td>1.102677</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1.261273</td>\n",
       "      <td>1.178617</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1234</td>\n",
       "      <td>1.070671</td>\n",
       "      <td>1.346140</td>\n",
       "      <td>1.063907</td>\n",
       "      <td>1.271996</td>\n",
       "      <td>1.897697</td>\n",
       "      <td>1.197229</td>\n",
       "      <td>1.143079</td>\n",
       "      <td>1.128461</td>\n",
       "      <td>2.173699</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.108208</td>\n",
       "      <td>1.012897</td>\n",
       "      <td>0.985740</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>2.098213</td>\n",
       "      <td>3.173915</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.301367</td>\n",
       "      <td>0.742789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 ANE_004_05M  ANE_004_40M  ANE_150_05M  ANE_150_40M  \\\n",
       "gene_callers_id                                                       \n",
       "1230                0.000000     0.000000     0.721253     0.000000   \n",
       "1231                0.000000     1.033706     1.632857     1.407635   \n",
       "1232                4.528344     7.415190     1.292512     3.206930   \n",
       "1233                0.000000     0.000000     1.376027     1.012331   \n",
       "1234                1.070671     1.346140     1.063907     1.271996   \n",
       "\n",
       "                 ANE_151_05M  ANE_151_80M  ANE_152_05M  ANW_141_05M  \\\n",
       "gene_callers_id                                                       \n",
       "1230                1.155447     1.070772     0.000000     1.307078   \n",
       "1231                0.909497     0.000000     0.000000     0.811527   \n",
       "1232                4.631825     5.682169     3.922132     3.784440   \n",
       "1233                1.112675     1.068535     0.935034     1.281728   \n",
       "1234                1.897697     1.197229     1.143079     1.128461   \n",
       "\n",
       "                 ANW_142_05M  ANW_145_05M  ...  PSW_128_40M  RED_31_05M  \\\n",
       "gene_callers_id                            ...                            \n",
       "1230                1.116991     0.000000  ...     0.789185    1.077167   \n",
       "1231                1.065848     0.759649  ...     0.000000    1.144542   \n",
       "1232                0.775314     1.938717  ...     2.257851    1.992068   \n",
       "1233                0.915704     0.872245  ...     0.941432    0.950875   \n",
       "1234                2.173699     0.000000  ...     1.108208    1.012897   \n",
       "\n",
       "                 RED_32_05M  RED_32_80M  RED_33_05M  RED_34_05M  RED_34_60M  \\\n",
       "gene_callers_id                                                               \n",
       "1230               2.026755     0.00000    0.959566    1.298346    0.971263   \n",
       "1231               0.642688     0.84632    1.154175    2.702909    1.713162   \n",
       "1232               4.024967     0.84488    2.544174    2.909985    1.034122   \n",
       "1233               1.102677     0.00000    1.261273    1.178617    0.000000   \n",
       "1234               0.985740     0.00000    2.098213    3.173915    0.000000   \n",
       "\n",
       "                 SOC_84_05M  SOC_85_05M  SOC_85_90M  \n",
       "gene_callers_id                                      \n",
       "1230               0.000000    0.000000    0.000000  \n",
       "1231               0.000000    0.000000    0.000000  \n",
       "1232               1.023991    1.681214    0.912291  \n",
       "1233               0.000000    0.000000    0.000000  \n",
       "1234               0.000000    0.301367    0.742789  \n",
       "\n",
       "[5 rows x 93 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sigma.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ANE_004_05M</th>\n",
       "      <th>ANE_004_40M</th>\n",
       "      <th>ANE_150_05M</th>\n",
       "      <th>ANE_150_40M</th>\n",
       "      <th>ANE_151_05M</th>\n",
       "      <th>ANE_151_80M</th>\n",
       "      <th>ANE_152_05M</th>\n",
       "      <th>ANW_141_05M</th>\n",
       "      <th>ANW_142_05M</th>\n",
       "      <th>ANW_145_05M</th>\n",
       "      <th>...</th>\n",
       "      <th>PSW_128_40M</th>\n",
       "      <th>RED_31_05M</th>\n",
       "      <th>RED_32_05M</th>\n",
       "      <th>RED_32_80M</th>\n",
       "      <th>RED_33_05M</th>\n",
       "      <th>RED_34_05M</th>\n",
       "      <th>RED_34_60M</th>\n",
       "      <th>SOC_84_05M</th>\n",
       "      <th>SOC_85_05M</th>\n",
       "      <th>SOC_85_90M</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>count</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "      <td>1472.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mean</td>\n",
       "      <td>39.038340</td>\n",
       "      <td>59.241591</td>\n",
       "      <td>57.084594</td>\n",
       "      <td>47.794651</td>\n",
       "      <td>57.749177</td>\n",
       "      <td>42.530901</td>\n",
       "      <td>22.053108</td>\n",
       "      <td>68.242723</td>\n",
       "      <td>49.498944</td>\n",
       "      <td>35.727333</td>\n",
       "      <td>...</td>\n",
       "      <td>47.610295</td>\n",
       "      <td>69.195827</td>\n",
       "      <td>68.153178</td>\n",
       "      <td>40.393658</td>\n",
       "      <td>83.733160</td>\n",
       "      <td>74.113797</td>\n",
       "      <td>17.808854</td>\n",
       "      <td>1.456135</td>\n",
       "      <td>1.685583</td>\n",
       "      <td>1.580930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>std</td>\n",
       "      <td>47.939791</td>\n",
       "      <td>54.767987</td>\n",
       "      <td>51.821394</td>\n",
       "      <td>44.160372</td>\n",
       "      <td>54.309530</td>\n",
       "      <td>39.572915</td>\n",
       "      <td>28.652507</td>\n",
       "      <td>59.733582</td>\n",
       "      <td>44.395447</td>\n",
       "      <td>31.658604</td>\n",
       "      <td>...</td>\n",
       "      <td>51.345438</td>\n",
       "      <td>61.335578</td>\n",
       "      <td>63.168394</td>\n",
       "      <td>35.491110</td>\n",
       "      <td>75.346165</td>\n",
       "      <td>65.319855</td>\n",
       "      <td>15.326817</td>\n",
       "      <td>9.921148</td>\n",
       "      <td>10.040211</td>\n",
       "      <td>10.643565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>min</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25%</td>\n",
       "      <td>10.598841</td>\n",
       "      <td>18.052542</td>\n",
       "      <td>17.340911</td>\n",
       "      <td>14.201948</td>\n",
       "      <td>17.588154</td>\n",
       "      <td>13.507111</td>\n",
       "      <td>5.844172</td>\n",
       "      <td>25.933356</td>\n",
       "      <td>11.327739</td>\n",
       "      <td>14.033472</td>\n",
       "      <td>...</td>\n",
       "      <td>5.611061</td>\n",
       "      <td>16.848131</td>\n",
       "      <td>18.132485</td>\n",
       "      <td>11.034903</td>\n",
       "      <td>26.539442</td>\n",
       "      <td>23.521712</td>\n",
       "      <td>5.491698</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50%</td>\n",
       "      <td>28.822122</td>\n",
       "      <td>49.254806</td>\n",
       "      <td>48.605339</td>\n",
       "      <td>39.575698</td>\n",
       "      <td>47.751805</td>\n",
       "      <td>35.425702</td>\n",
       "      <td>15.121102</td>\n",
       "      <td>55.181912</td>\n",
       "      <td>41.231201</td>\n",
       "      <td>29.423429</td>\n",
       "      <td>...</td>\n",
       "      <td>33.473685</td>\n",
       "      <td>59.120863</td>\n",
       "      <td>55.029042</td>\n",
       "      <td>34.827187</td>\n",
       "      <td>69.758874</td>\n",
       "      <td>62.221538</td>\n",
       "      <td>15.689128</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75%</td>\n",
       "      <td>52.056984</td>\n",
       "      <td>83.406521</td>\n",
       "      <td>83.349484</td>\n",
       "      <td>68.880770</td>\n",
       "      <td>82.554984</td>\n",
       "      <td>61.074302</td>\n",
       "      <td>28.653265</td>\n",
       "      <td>96.089626</td>\n",
       "      <td>71.493395</td>\n",
       "      <td>50.444486</td>\n",
       "      <td>...</td>\n",
       "      <td>71.061427</td>\n",
       "      <td>104.886325</td>\n",
       "      <td>98.808691</td>\n",
       "      <td>58.766579</td>\n",
       "      <td>119.525505</td>\n",
       "      <td>103.623128</td>\n",
       "      <td>25.622690</td>\n",
       "      <td>0.856785</td>\n",
       "      <td>0.906532</td>\n",
       "      <td>0.959329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>max</td>\n",
       "      <td>710.677520</td>\n",
       "      <td>575.668786</td>\n",
       "      <td>594.067943</td>\n",
       "      <td>506.453967</td>\n",
       "      <td>632.177044</td>\n",
       "      <td>503.727956</td>\n",
       "      <td>441.194259</td>\n",
       "      <td>442.848919</td>\n",
       "      <td>278.286786</td>\n",
       "      <td>444.704508</td>\n",
       "      <td>...</td>\n",
       "      <td>445.336367</td>\n",
       "      <td>363.562762</td>\n",
       "      <td>424.735719</td>\n",
       "      <td>229.567542</td>\n",
       "      <td>488.720994</td>\n",
       "      <td>375.455102</td>\n",
       "      <td>136.135438</td>\n",
       "      <td>316.719406</td>\n",
       "      <td>334.981276</td>\n",
       "      <td>302.703189</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ANE_004_05M  ANE_004_40M  ANE_150_05M  ANE_150_40M  ANE_151_05M  \\\n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000   \n",
       "mean     39.038340    59.241591    57.084594    47.794651    57.749177   \n",
       "std      47.939791    54.767987    51.821394    44.160372    54.309530   \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000   \n",
       "25%      10.598841    18.052542    17.340911    14.201948    17.588154   \n",
       "50%      28.822122    49.254806    48.605339    39.575698    47.751805   \n",
       "75%      52.056984    83.406521    83.349484    68.880770    82.554984   \n",
       "max     710.677520   575.668786   594.067943   506.453967   632.177044   \n",
       "\n",
       "       ANE_151_80M  ANE_152_05M  ANW_141_05M  ANW_142_05M  ANW_145_05M  ...  \\\n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000  ...   \n",
       "mean     42.530901    22.053108    68.242723    49.498944    35.727333  ...   \n",
       "std      39.572915    28.652507    59.733582    44.395447    31.658604  ...   \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
       "25%      13.507111     5.844172    25.933356    11.327739    14.033472  ...   \n",
       "50%      35.425702    15.121102    55.181912    41.231201    29.423429  ...   \n",
       "75%      61.074302    28.653265    96.089626    71.493395    50.444486  ...   \n",
       "max     503.727956   441.194259   442.848919   278.286786   444.704508  ...   \n",
       "\n",
       "       PSW_128_40M   RED_31_05M   RED_32_05M   RED_32_80M   RED_33_05M  \\\n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000   \n",
       "mean     47.610295    69.195827    68.153178    40.393658    83.733160   \n",
       "std      51.345438    61.335578    63.168394    35.491110    75.346165   \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000   \n",
       "25%       5.611061    16.848131    18.132485    11.034903    26.539442   \n",
       "50%      33.473685    59.120863    55.029042    34.827187    69.758874   \n",
       "75%      71.061427   104.886325    98.808691    58.766579   119.525505   \n",
       "max     445.336367   363.562762   424.735719   229.567542   488.720994   \n",
       "\n",
       "        RED_34_05M   RED_34_60M   SOC_84_05M   SOC_85_05M   SOC_85_90M  \n",
       "count  1472.000000  1472.000000  1472.000000  1472.000000  1472.000000  \n",
       "mean     74.113797    17.808854     1.456135     1.685583     1.580930  \n",
       "std      65.319855    15.326817     9.921148    10.040211    10.643565  \n",
       "min       0.000000     0.000000     0.000000     0.000000     0.000000  \n",
       "25%      23.521712     5.491698     0.000000     0.000000     0.000000  \n",
       "50%      62.221538    15.689128     0.000000     0.000000     0.000000  \n",
       "75%     103.623128    25.622690     0.856785     0.906532     0.959329  \n",
       "max     375.455102   136.135438   316.719406   334.981276   302.703189  \n",
       "\n",
       "[8 rows x 93 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sigma.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We ideally want to get a good count of the number of genes in each metagenome, so we use the count function as shown below. Since df_sigma has the same number of entries, counting its values would be redundant and is therefore omitted in the notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ANE_004_05M    1472\n",
       "ANE_004_40M    1472\n",
       "ANE_150_05M    1472\n",
       "ANE_150_40M    1472\n",
       "ANE_151_05M    1472\n",
       "               ... \n",
       "RED_34_05M     1472\n",
       "RED_34_60M     1472\n",
       "SOC_84_05M     1472\n",
       "SOC_85_05M     1472\n",
       "SOC_85_90M     1472\n",
       "Length: 93, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.count()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we want to analyze the coverage values for each metagenome in the sampling environment. The goal is to split each column from df and then classify the genes based on the coverage value. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "ename": "IndentationError",
     "evalue": "unexpected indent (<ipython-input-24-ba186a08937d>, line 10)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-24-ba186a08937d>\"\u001b[0;36m, line \u001b[0;32m10\u001b[0m\n\u001b[0;31m    for column in fixed_file.columns:\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mIndentationError\u001b[0m\u001b[0;31m:\u001b[0m unexpected indent\n"
     ]
    }
   ],
   "source": [
    "def split_metagenome_values(file, desired_column):\n",
    "    desired = int(desired_column)\n",
    "    length = len(file)\n",
    "    columns = []\n",
    "    fixed_file = file.reset_index(drop=True)\n",
    "    \n",
    "    for column in fixed_file.columns:\n",
    "        separate_column = file['{}'.format(column)]\n",
    "        columns.append(separate_column)\n",
    "    \n",
    "    columns = np.array_split(columns, len(columns))\n",
    "    \n",
    "    if desired > length:\n",
    "        raise Exception(\"Error! You inputted {} but the maximum number of columns available are {}\"\n",
    "                        .format(desired, length))\n",
    "        \n",
    "    return columns[(desired - 1)] \n",
    "\n",
    "def rearrange_columns(file):\n",
    "    values = {}\n",
    "    fixed_file = file.reset_index(drop=True)\n",
    "    column_values = split_column_values(file)\n",
    "    \n",
    "    \n",
    "    for column in fixed_file.columns:\n",
    "        values['{}'.format(column)] = split_column_values[fixed_file.index(column)]\n",
    "        print(column)\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[  0.           0.           2.30820399 ...   1.14003591 460.27594937\n",
      "  135.91421947]]\n"
     ]
    }
   ],
   "source": [
    "print(split_metagenome_values(df, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The number of absent genes in this metagenome is 93\n"
     ]
    }
   ],
   "source": [
    "def absent_genes(file): # Returns the number of absent genes in each environment\n",
    "    x = {}\n",
    "    for column in file.columns:\n",
    "        if column == file.index:\n",
    "            pass\n",
    "        else:\n",
    "            query = file.loc[file['{}'.format(column)] <= 100].count()[0]\n",
    "            x['{}'.format(column)] = '{}'.format(query)\n",
    "    return x \n",
    "\n",
    "def total_number_of_absent_genes(file):\n",
    "    count = 0\n",
    "    length = len(absent_genes(file))\n",
    "    return \"The number of absent genes in this metagenome is {}\".format(length)\n",
    "\n",
    "print(number_of_absent_genes(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'Series' objects are mutable, thus they cannot be hashed",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-112-30e0e428bfc3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mmetagenomes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseparate_metagenomes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-112-30e0e428bfc3>\u001b[0m in \u001b[0;36mseparate_metagenomes\u001b[0;34m(file)\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0mfile_columns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdrop\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfile_columns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m         \u001b[0mi\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmetagenomes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'{}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mmetagenomes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__hash__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1884\u001b[0m         raise TypeError(\n\u001b[1;32m   1885\u001b[0m             \u001b[0;34m\"{0!r} objects are mutable, thus they cannot be\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1886\u001b[0;31m             \u001b[0;34m\" hashed\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__class__\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1887\u001b[0m         )\n\u001b[1;32m   1888\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: 'Series' objects are mutable, thus they cannot be hashed"
     ]
    }
   ],
   "source": [
    "def separate_metagenomes(file):\n",
    "    metagenomes = {}\n",
    "    file_columns = file.reset_index(drop = True)\n",
    "    for i in file_columns.columns:\n",
    "        i = metagenomes[file['{}'.format(i)]]\n",
    "    return metagenomes\n",
    "    \n",
    "print(separate_metagenomes(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ANE_004_05M</th>\n",
       "      <th>ANE_004_40M</th>\n",
       "      <th>ANE_150_05M</th>\n",
       "      <th>ANE_150_40M</th>\n",
       "      <th>ANE_151_05M</th>\n",
       "      <th>ANE_151_80M</th>\n",
       "      <th>ANE_152_05M</th>\n",
       "      <th>ANW_141_05M</th>\n",
       "      <th>ANW_142_05M</th>\n",
       "      <th>ANW_145_05M</th>\n",
       "      <th>...</th>\n",
       "      <th>PSW_128_40M</th>\n",
       "      <th>RED_31_05M</th>\n",
       "      <th>RED_32_05M</th>\n",
       "      <th>RED_32_80M</th>\n",
       "      <th>RED_33_05M</th>\n",
       "      <th>RED_34_05M</th>\n",
       "      <th>RED_34_60M</th>\n",
       "      <th>SOC_84_05M</th>\n",
       "      <th>SOC_85_05M</th>\n",
       "      <th>SOC_85_90M</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.701389</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.955056</td>\n",
       "      <td>1.082962</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.044402</td>\n",
       "      <td>1.196158</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.959543</td>\n",
       "      <td>0.920139</td>\n",
       "      <td>3.242754</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.801282</td>\n",
       "      <td>7.103571</td>\n",
       "      <td>0.885068</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.779553</td>\n",
       "      <td>1.075638</td>\n",
       "      <td>1.009675</td>\n",
       "      <td>0.640091</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.894249</td>\n",
       "      <td>1.137011</td>\n",
       "      <td>0.745358</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.985728</td>\n",
       "      <td>0.502580</td>\n",
       "      <td>0.631198</td>\n",
       "      <td>0.895377</td>\n",
       "      <td>3.016057</td>\n",
       "      <td>1.743184</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>2.308204</td>\n",
       "      <td>8.848606</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>2.940000</td>\n",
       "      <td>4.534161</td>\n",
       "      <td>4.615087</td>\n",
       "      <td>3.880342</td>\n",
       "      <td>4.780449</td>\n",
       "      <td>0.566667</td>\n",
       "      <td>1.901376</td>\n",
       "      <td>...</td>\n",
       "      <td>3.362563</td>\n",
       "      <td>2.746753</td>\n",
       "      <td>5.094574</td>\n",
       "      <td>0.656842</td>\n",
       "      <td>6.289425</td>\n",
       "      <td>6.549606</td>\n",
       "      <td>0.717694</td>\n",
       "      <td>1.008104</td>\n",
       "      <td>1.869355</td>\n",
       "      <td>1.077206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.673582</td>\n",
       "      <td>1.717138</td>\n",
       "      <td>5.044211</td>\n",
       "      <td>1.370543</td>\n",
       "      <td>1.016949</td>\n",
       "      <td>1.592806</td>\n",
       "      <td>1.721739</td>\n",
       "      <td>0.838174</td>\n",
       "      <td>...</td>\n",
       "      <td>1.092669</td>\n",
       "      <td>0.920128</td>\n",
       "      <td>2.012346</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.829588</td>\n",
       "      <td>4.966741</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>0.370583</td>\n",
       "      <td>1.080229</td>\n",
       "      <td>1.181025</td>\n",
       "      <td>1.667647</td>\n",
       "      <td>2.080544</td>\n",
       "      <td>2.748344</td>\n",
       "      <td>0.713467</td>\n",
       "      <td>2.011351</td>\n",
       "      <td>3.738072</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.802871</td>\n",
       "      <td>1.115991</td>\n",
       "      <td>1.375427</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>9.217054</td>\n",
       "      <td>7.203260</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.086915</td>\n",
       "      <td>0.203438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1467</td>\n",
       "      <td>94.403433</td>\n",
       "      <td>163.625920</td>\n",
       "      <td>191.183499</td>\n",
       "      <td>156.819225</td>\n",
       "      <td>171.202050</td>\n",
       "      <td>129.765648</td>\n",
       "      <td>29.571637</td>\n",
       "      <td>383.258491</td>\n",
       "      <td>125.886331</td>\n",
       "      <td>137.729462</td>\n",
       "      <td>...</td>\n",
       "      <td>62.136966</td>\n",
       "      <td>299.221277</td>\n",
       "      <td>298.376420</td>\n",
       "      <td>112.908148</td>\n",
       "      <td>439.994169</td>\n",
       "      <td>418.955587</td>\n",
       "      <td>52.779710</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.017544</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1468</td>\n",
       "      <td>79.135802</td>\n",
       "      <td>98.962963</td>\n",
       "      <td>106.032099</td>\n",
       "      <td>80.679012</td>\n",
       "      <td>107.022222</td>\n",
       "      <td>75.180247</td>\n",
       "      <td>56.246914</td>\n",
       "      <td>45.904215</td>\n",
       "      <td>34.250871</td>\n",
       "      <td>81.133333</td>\n",
       "      <td>...</td>\n",
       "      <td>70.977778</td>\n",
       "      <td>19.752988</td>\n",
       "      <td>17.428571</td>\n",
       "      <td>12.689394</td>\n",
       "      <td>49.453488</td>\n",
       "      <td>61.388889</td>\n",
       "      <td>34.249383</td>\n",
       "      <td>10.589189</td>\n",
       "      <td>7.917293</td>\n",
       "      <td>13.508642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1469</td>\n",
       "      <td>1.140036</td>\n",
       "      <td>3.108664</td>\n",
       "      <td>1.180617</td>\n",
       "      <td>0.911179</td>\n",
       "      <td>3.945017</td>\n",
       "      <td>2.908955</td>\n",
       "      <td>0.728341</td>\n",
       "      <td>1.962519</td>\n",
       "      <td>0.679883</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.663730</td>\n",
       "      <td>0.888399</td>\n",
       "      <td>4.034314</td>\n",
       "      <td>0.881288</td>\n",
       "      <td>5.787313</td>\n",
       "      <td>3.299803</td>\n",
       "      <td>0.933635</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1470</td>\n",
       "      <td>460.275949</td>\n",
       "      <td>477.937019</td>\n",
       "      <td>425.980569</td>\n",
       "      <td>445.023185</td>\n",
       "      <td>452.983209</td>\n",
       "      <td>327.917096</td>\n",
       "      <td>304.368210</td>\n",
       "      <td>440.024780</td>\n",
       "      <td>369.529720</td>\n",
       "      <td>307.220227</td>\n",
       "      <td>...</td>\n",
       "      <td>366.529826</td>\n",
       "      <td>527.093207</td>\n",
       "      <td>384.422091</td>\n",
       "      <td>301.627179</td>\n",
       "      <td>456.171693</td>\n",
       "      <td>401.840067</td>\n",
       "      <td>226.718171</td>\n",
       "      <td>21.054922</td>\n",
       "      <td>18.620078</td>\n",
       "      <td>20.466807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1471</td>\n",
       "      <td>135.914219</td>\n",
       "      <td>165.365909</td>\n",
       "      <td>152.231431</td>\n",
       "      <td>126.879242</td>\n",
       "      <td>160.493477</td>\n",
       "      <td>116.465294</td>\n",
       "      <td>112.599388</td>\n",
       "      <td>91.268755</td>\n",
       "      <td>129.693309</td>\n",
       "      <td>120.004345</td>\n",
       "      <td>...</td>\n",
       "      <td>90.066116</td>\n",
       "      <td>111.068447</td>\n",
       "      <td>73.503791</td>\n",
       "      <td>76.984707</td>\n",
       "      <td>101.597458</td>\n",
       "      <td>90.096854</td>\n",
       "      <td>62.472792</td>\n",
       "      <td>12.176353</td>\n",
       "      <td>13.472603</td>\n",
       "      <td>11.876786</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1472 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ANE_004_05M  ANE_004_40M  ANE_150_05M  ANE_150_40M  ANE_151_05M  \\\n",
       "0        0.000000     0.000000     0.701389     0.000000     2.955056   \n",
       "1        0.000000     0.779553     1.075638     1.009675     0.640091   \n",
       "2        2.308204     8.848606     0.800000     2.940000     4.534161   \n",
       "3        0.000000     0.000000     1.673582     1.717138     5.044211   \n",
       "4        0.370583     1.080229     1.181025     1.667647     2.080544   \n",
       "...           ...          ...          ...          ...          ...   \n",
       "1467    94.403433   163.625920   191.183499   156.819225   171.202050   \n",
       "1468    79.135802    98.962963   106.032099    80.679012   107.022222   \n",
       "1469     1.140036     3.108664     1.180617     0.911179     3.945017   \n",
       "1470   460.275949   477.937019   425.980569   445.023185   452.983209   \n",
       "1471   135.914219   165.365909   152.231431   126.879242   160.493477   \n",
       "\n",
       "      ANE_151_80M  ANE_152_05M  ANW_141_05M  ANW_142_05M  ANW_145_05M  ...  \\\n",
       "0        1.082962     0.000000     2.044402     1.196158     0.000000  ...   \n",
       "1        0.000000     0.000000     0.894249     1.137011     0.745358  ...   \n",
       "2        4.615087     3.880342     4.780449     0.566667     1.901376  ...   \n",
       "3        1.370543     1.016949     1.592806     1.721739     0.838174  ...   \n",
       "4        2.748344     0.713467     2.011351     3.738072     0.000000  ...   \n",
       "...           ...          ...          ...          ...          ...  ...   \n",
       "1467   129.765648    29.571637   383.258491   125.886331   137.729462  ...   \n",
       "1468    75.180247    56.246914    45.904215    34.250871    81.133333  ...   \n",
       "1469     2.908955     0.728341     1.962519     0.679883     0.000000  ...   \n",
       "1470   327.917096   304.368210   440.024780   369.529720   307.220227  ...   \n",
       "1471   116.465294   112.599388    91.268755   129.693309   120.004345  ...   \n",
       "\n",
       "      PSW_128_40M  RED_31_05M  RED_32_05M  RED_32_80M  RED_33_05M  RED_34_05M  \\\n",
       "0        0.959543    0.920139    3.242754    0.000000    1.801282    7.103571   \n",
       "1        0.000000    0.985728    0.502580    0.631198    0.895377    3.016057   \n",
       "2        3.362563    2.746753    5.094574    0.656842    6.289425    6.549606   \n",
       "3        1.092669    0.920128    2.012346    0.000000    3.829588    4.966741   \n",
       "4        1.802871    1.115991    1.375427    0.000000    9.217054    7.203260   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "1467    62.136966  299.221277  298.376420  112.908148  439.994169  418.955587   \n",
       "1468    70.977778   19.752988   17.428571   12.689394   49.453488   61.388889   \n",
       "1469     0.663730    0.888399    4.034314    0.881288    5.787313    3.299803   \n",
       "1470   366.529826  527.093207  384.422091  301.627179  456.171693  401.840067   \n",
       "1471    90.066116  111.068447   73.503791   76.984707  101.597458   90.096854   \n",
       "\n",
       "      RED_34_60M  SOC_84_05M  SOC_85_05M  SOC_85_90M  \n",
       "0       0.885068    0.000000    0.000000    0.000000  \n",
       "1       1.743184    0.000000    0.000000    0.000000  \n",
       "2       0.717694    1.008104    1.869355    1.077206  \n",
       "3       0.000000    0.000000    0.000000    0.000000  \n",
       "4       0.000000    0.000000    0.086915    0.203438  \n",
       "...          ...         ...         ...         ...  \n",
       "1467   52.779710    0.000000    1.017544    0.000000  \n",
       "1468   34.249383   10.589189    7.917293   13.508642  \n",
       "1469    0.933635    0.000000    0.000000    0.000000  \n",
       "1470  226.718171   21.054922   18.620078   20.466807  \n",
       "1471   62.472792   12.176353   13.472603   11.876786  \n",
       "\n",
       "[1472 rows x 93 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.reset_index(drop = True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
